File: exts/jphp-gui-ext/src/main/java/org/develnext/jphp/ext/javafx/classes/layout/UXPane.java
package org.develnext.jphp.ext.javafx.classes.layout;

import javafx.collections.ObservableList;
import javafx.scene.Node;
import javafx.scene.layout.Pane;
import org.develnext.jphp.ext.javafx.JavaFXExtension;
import org.develnext.jphp.ext.javafx.support.control.tabpane.DndTabPane;
import php.runtime.annotation.Reflection.Getter;
import php.runtime.annotation.Reflection.Name;
import php.runtime.annotation.Reflection.Signature;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

@Name(JavaFXExtension.NS + "layout\\UXPane")
public class UXPane<T extends Pane> extends UXRegion<T> {
    interface WrappedInterface {

    }

    public UXPane(Environment env, T wrappedObject) {
        super(env, wrappedObject);
    }

    public UXPane(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Override
    public T getWrappedObject() {
        return (T) super.getWrappedObject();
    }

    @Getter
    public ObservableList<Node> getChildren() {
        return getWrappedObject().getChildren();
    }

    @Signature
    public void __construct() {
        __wrappedObject = new Pane();
    }

    @Signature
    public void add(Node node) {
        getChildren().add(node);
    }

    @Signature
    public boolean remove(Node node) {
        getWrappedObject().getPrefWidth();
        return getChildren().remove(node);
    }
}

File: exts/jphp-gui-ext/src/main/resources/JPHP-INF/sdk/php/gui/layout/UXPane.php
<?php
namespace php\gui\layout;

use php\gui\UXList;
use php\gui\UXNode;
use php\gui\UXParent;

/**
 * Class UXPane
 * @package php\gui
 * @packages gui, javafx
 */
class UXPane extends UXRegion
{
    /**
     * @readonly
     * @var UXList
     */
    public $children;

    /**
     * @param UXNode $node
     */
    public function add(UXNode $node)
    {
    }

    /**
     * @param UXNode $node
     */
    public function remove(UXNode $node)
    {
    }
}